### PR TITLE
Helm: default to ConfigMap in configStorageType value

### LINKED ISSUE #2018: Helm: switch default storage of mimir config from secret to configmap
This is a breaking change, so bump major chart version as needed.

Final step in the series of issues opened around configuration handling. See tracking EPIC , should come after https://github.com/grafana/helm-charts/issues/1292

### PR DESCRIPTION
**Pull request by krajorama**
#### What this PR does

Flip the default.
Last step in the configuration improvements.

#### Which issue(s) this PR fixes or relates to

Fixes #2018 

#### Checklist

- [x] Tests updated
- [N/A] Documentation added - done separately https://github.com/grafana/mimir/issues/2172
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ Entries should be ordered as follows:
 Entries should include a reference to the Pull Request that introduced the change.
 
 ## main / unreleased
+* [CHANGE] **breaking change** Make `ConfigMap` the default for `configStorageType`. This means that the Mimir (or Enterprise Metrics) configuration is now created in and loaded from a ConfigMap instead of a Secret. #2277
+  - Set to `Secret` to keep existing way of working. See related #2031, #2017, #2089.
```
Comment (jhesketh) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:16]: Is there any guide on how to migrate to a ConfigMap?

**Review by jhesketh — COMMENTED** [review]
Change lgtm; just a query regarding the break-ness of this.

File: operations/helm/charts/mimir-distributed/CHANGELOG.md (hunk 1 above)
Comment (krajorama) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md:16]: No, there's not :(

**Review by osg-grafana — APPROVED** [review]

**Comment by osg-grafana** [comment]
Removed `type/docs` label because there is a separate doc issue.

**Review by replay — APPROVED** [review]
LGTM, thanks!

**Pull request merged**